### ISSUE TITLE
Clicking on a sub-resource closes the sub-inspector it is located in, instead of opening in a sub-inspector, if an EditorPlugin `handles` it

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#64700]
### Godot version

f2a61684143af02a8cbe7002645af817607f9bd6

### System information

Windows 10 64 bits GLES3 GeForce GTX 1060 6GB/PCIe/SSE2

### Issue description

I wanted to adjust some inspector UI in my module and got faced with a strange unexpected behavior.

I have the following properties in a custom node:

![image](https://user-images.githubusercontent.com/1311555/185804917-9fa6852f-6ae5-4695-897f-ff7276b51ca9.png)

`library` is a resource, which contains sub-resources. 
If I click on the sub-resource on the right of `Item 1`, it usually opens a nested sub-inspector. However, if I create an `EditorPlugin` which `handles` the class of that sub-resource, clicking that sub-resource instead *closes the parent sub-inspector*, leaving it in this state:

![image](https://user-images.githubusercontent.com/1311555/185804979-2a1bc610-186a-468f-8a59-beb6e84fe24b.png)

This happens even if the editor plugin does nothing, apart from returning `true` in `handles` for that class.

### Steps to reproduce

Have `MyNode` with a `MyResource` property. `MyResource` in turn contains a `MySubResource` property. Then have two editor plugins, one handling `MyResource`, and another handling `MySubResource`. They don't need to do anything particular, just return `true` when matching their associated class in the `handles` method.

### Minimal reproduction project

The reproduction steps are quite specific so I wrote a reproduction module, which you can paste in the `modules` folder:
[subinspector_plugin_issue.zip](https://github.com/godotengine/godot/files/9389725/subinspector_plugin_issue.zip)

Create a `MyNode`, then in the inspector of that node, create a new `MyResource`. Then inside the `MyResource`, create a new `MySubResource`. Observe clicking on `MySubResource` closes the `MyResource` inspector instead of opening a `MySubResource` inspector.

**Comment by YuriSizov** [comment]
This is a duplicate of https://github.com/godotengine/godot/issues/52779. At one point a proper solution was supposed to be https://github.com/godotengine/godot/pull/45085, but @reduz preferred to fix the plugins we have built-in and change their behavior to avoid these issues. Of course, this doesn't work for user plugins.

In 3.x this is mitigated by disabling sub-inspectors for sub-resources in such cases: https://github.com/godotengine/godot/pull/52984.

**Comment by Zylann** [comment]
Right, in this present case I actually didn't need to use `handles()` since the functionality is confined to the inspector, there is no other particular editor popping up. I was just confused about the unexpected behavior and figured I'd write my thoughts about it. So technically I wasn't blocked yet, I hope it gets figured out.

**Comment by Bonkahe** [comment]
Still experiencing this in 2025, sorry to necro a very old issue, but evidently it is still a problem.
